spyder/plugins/explorer/widgets/fileassociations.py

```diff
@@ -0,0 +1,525 @@
+        self.list_extensions.currentRowChanged.connect(self.update_extensions)
+        self.list_extensions.itemDoubleClicked.connect(self.edit_association)
+        self.list_applications.currentRowChanged.connect(
+            self.update_applications)
+        self._refresh()
+        self._create_association_dialog()
+
+    def _refresh(self):
+        """Refresh the status of buttons on widget."""
+        self.setUpdatesEnabled(False)
+        for widget in [self.button_remove,  self.button_add_application,
+                       self.button_edit,
+                       self.button_remove_application, self.button_default]:
+            widget.setDisabled(True)
+
+        item = self.list_extensions.currentItem()
+        if item:
+            for widget in [self.button_remove, self.button_add_application,
+                           self.button_remove_application, self.button_edit]:
+                widget.setDisabled(False)
+        self.setUpdatesEnabled(True)
+
+    def _add_association(self, value):
+        """Add association helper."""
+        # Check value is not pressent
+        for row in range(self.list_extensions.count()):
+            item = self.list_extensions.item(row)
+            if item.text().strip() == value.strip():
+                break
+        else:
+            item = QListWidgetItem(value)
+            self.list_extensions.addItem(item)
+            self.list_extensions.setCurrentItem(item)
+
+        self._refresh()
+
+    def _add_application(self, app_name, fpath):
+        """Add application helper."""
+        for row in range(self.list_applications.count()):
+            item = self.list_extensions.item(row)
+            if item and item.text().strip() == app_name:
+                break
+        else:
+            icon = get_application_icon(fpath)
+
+            if not (os.path.isfile(fpath) or os.path.isdir(fpath)):
+                app_name += _(' (Application not found!)')
+
+            item = QListWidgetItem(icon, app_name)
+            self.list_applications.addItem(item)
+            self.list_applications.setCurrentItem(item)
+
+        if not (os.path.isfile(fpath) or os.path.isdir(fpath)):
+            item.setToolTip(_('Application not found!'))
+
+    def _update_extensions(self):
+        """Update extensions list."""
+        self.list_extensions.clear()
+        for extension, _ in sorted(self._data.items()):
+            self._add_association(extension)
+
+        # Select first item
+        self.list_extensions.setCurrentRow(0)
+        self.update_extensions()
+        self.update_applications()
+
+    def _create_association_dialog(self):
+        """Create input extension dialog and save it to for reuse."""
+        self._dlg_input = InputTextDialog(
+            self,
+            title=_('File association'),
+            label=(
+                _('Enter new file association/extension')
+                + ' (e.g. <code>*.txt</code> or <code>name.ext</code>)'
```

Comment:
`name.ext` is confuse. I think we should leave only `*.txt` or `.json`.

Reply:
@ccordoba12 the idea is that there might be a case for filenames that are fixed. Let's say I want `MANIFEST.in` files to open using a specific extension and not just `*.in`

How could I word this better?

Reply:
Perhaps we should give a concrete example, i.e. instead of `name.ext` you could use `README.md` or `CHANGELOG`.